text/1164034/phase_king.tex

```diff
@@ -0,0 +1,78 @@
+\documentclass[11pt, a4paper]{report}
+\usepackage[utf8]{inputenc}
+\usepackage{amsmath}
+\usepackage{amssymb}
+% \usepackage{polski}
+\usepackage{fancyhdr}
+\usepackage{lastpage}
+\usepackage{relsize}
+\usepackage{listings}
+\usepackage{mathtools}
+\usepackage{array}
+\usepackage{multirow}
+\usepackage{tabularx}
+\usepackage{placeins}
+\usepackage{paralist}
+\usepackage{xcolor}
+\usepackage{amsthm}
+\usepackage{float}
+\usepackage{hyperref}
+\usepackage[normalem]{ulem}
+\usepackage{blindtext}
+
+\usepackage[
+    bindingoffset=-15pt,
+	voffset=0pt, 
+	hoffset=10pt, 
+	textwidth=500pt, 
+    textheight=650pt, 
+	marginparwidth=20pt,
+	footskip=60pt
+]{geometry}
+\pagestyle{fancy}
+
+\headheight=28pt
+
+\renewcommand{\footrulewidth}{0.4pt}
+
+
+\begin{document}
+
+\lhead{Mateusz Pach}
+\chead{\textbf{King Phase Protocol}}
+\rhead{\today{}}
+\cfoot{Page \thepage/\pageref{LastPage}}
+
+\theoremstyle{plain}
+\newtheorem{thm}{Theorem} % reset theorem numbering for each chapter
+\newtheorem{lem}{Lemma} % definition numbers are dependent on theorem numbers
+\newtheorem{exmp}{Example}% same for example numbers
+\theoremstyle{definition}
+\newtheorem{defn}{Definition} % definition numbers are dependent on theorem numbers
+
+\subsection*{Introduction} 
+Let us remind the consensus problem.
+\begin{defn}[The consensus problem]
```

Comment:
In fact it's a harder problem than consensus, where we allow byzantine nodes - and for such setting the problem is more precisely called "byzantine agreement", not "consensus". I know that for synchronous networks a simpler consensus problems (with failing nodes) does not have much sense, but please change this or at least include a comment on this.